frontend/src/metabase/ui/components/inputs/TextInputBlurChange/TextInputBlurChange.tsx

```diff
@@ -59,6 +59,21 @@ export function TextInputBlurChange<T extends TextInputProps = TextInputProps>({
     [normalize, onBlur, onBlurChange, value],
   );
 
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === "Escape") {
+        setInternalValue(value);
+
+        // setTimeout to allow internalValue to update before dispatching blur event
+        // so that blur event's target will have proper value
+        setTimeout(() => {
+          ref.current?.blur();
+        }, 0);
```

Comment:
Could manually setting the `ref.current.value` before calling `blur()` work too?
It'd be great to avoid the `setTimeout`.

Reply:
237083d6358711e0731a78302c639c0a756eb304